[BUG] api 目录冲突导致工作流失败

## 🐞 描述 Bug

  文档部署工作流 (`docs-deployment.yml`) 在"合并文档到统一目录"步骤中失败，错误信息为：
  cp: cannot overwrite non-directory 'site/api/os' with directory 'os/target/riscv64gc-unknown-none-elf/doc/os'
  Error: Process completed with exit code 1.

  根本原因是 mdBook 构建时根据 `SUMMARY.md` 中的 `[API 文档](api/os)` 链接生成了一个文件（`book/api/os` 或 `api/os.html`），而工作流后续尝试将 cargo doc 生成的
  `os` 目录复制到同一位置，导致冲突。

  ## ⚙️ 复现步骤

  提供重现此 Bug 所需的清晰、详细的步骤：

  1. 推送代码到 main 分支，触发文档部署工作流
  2. 观察工作流执行到"合并文档到统一目录"步骤
  3. 看到错误：`cp: cannot overwrite non-directory 'site/api/os' with directory 'os/target/riscv64gc-unknown-none-elf/doc/os'`

  或者本地复现：

  1. 运行 `mdbook build`（会生成 `book/api/os` 相关文件）
  2. 运行 `cd os && cargo doc --no-deps --target riscv64gc-unknown-none-elf`
  3. 执行工作流中的合并命令：
     ```bash
     mkdir -p site
     cp -r book/* site/
     mkdir -p site/api
     cp -r os/target/riscv64gc-unknown-none-elf/doc/* site/api/
  4. 看到相同的错误

  预期行为

  文档部署工作流应该成功完成，将 mdBook 生成的文档和 cargo doc 生成的 API 文档正确合并到 site/ 目录中。

  实际行为

  工作流在复制 API 文档时失败，因为无法用目录覆盖已存在的文件。

  🖥️ 环境信息

  - 目标架构 (Target Architecture): RISC-V 64
  - 模拟器/硬件平台 (Emulator/Hardware): GitHub Actions (ubuntu-latest)
  - 宿主机操作系统 (Host OS): Ubuntu (GitHub Actions runner)
  - 编译工具链版本 (Toolchain Version): Rust nightly-2025-10-28, target riscv64gc-unknown-none-elf
  - 代码提交哈希 (Commit Hash): effbcd8

  📝 截图或日志 (可选)

  错误日志：
  Run mkdir -p site
    mkdir -p site
    # 复制 mdBook 输出
    cp -r book/* site/
    # 复制 cargo doc 输出到 api 子目录
    mkdir -p site/api
    cp -r os/target/riscv64gc-unknown-none-elf/doc/* site/api/
    shell: /usr/bin/bash -e {0}
  cp: cannot overwrite non-directory 'site/api/os' with directory 'os/target/riscv64gc-unknown-none-elf/doc/os'
  Error: Process completed with exit code 1.

  💡 建议的解决方案

  在复制 cargo doc 输出前，先删除 mdBook 生成的 site/api 目录

--- a/.github/workflows/docs-deployment.yml
+++ b/.github/workflows/docs-deployment.yml
@@ -55,6 +55,8 @@ jobs:
           mkdir -p site
           # 复制 mdBook 输出
           cp -r book/* site/
+          # 删除 mdBook 生成的 api 目录（如果存在），避免与 cargo doc 输出冲突
+          rm -rf site/api
           # 复制 cargo doc 输出到 api 子目录
           mkdir -p site/api
           cp -r os/target/riscv64gc-unknown-none-elf/doc/* site/api/

